src/perftest_communication.c

```diff
@@ -938,7 +947,12 @@ int rdma_client_connect(struct pingpong_context *ctx,struct perftest_parameters
 		}
 		memset(&source_sin, 0x0, sizeof(source_sin));
 		//coverity[deref_after_free]
-		memcpy(&source_sin, res->ai_addr, sizeof(source_sin));
+		if (((struct sockaddr *)&sin)->sa_family == AF_INET) {
```

Comment:
this line will work, but `sin` is not related to code around here, better to use `res->ai_addr->sa_family == AF_INET`

Reply:
fixed